<launch>
      <arg name="rate" default="10"/>

      <arg name="ip"  default="172.29.32.1"/>

      <arg name="toggle_drone" default="0" />

      <arg name="vehicle_name" default="airsimvehicle" />

      <arg name="pose_topic" default="airsim/gtpose" />
      <arg name="pose_frame" default="world" />

      <arg name="vehicle_base_frame" default="base_link" />

      <arg name="sensor_imu_enable" default="1" />
      <arg name="sensor_imu_name" default="imu" />
      <arg name="sensor_imu_topic" default="airsim/imu" />
      <arg name="sensor_imu_frame" default="imu_link" />

      <node name="airsim_record_route" pkg="airsim" type="airsim_record_route.py" respawn="false" output="screen">

            <rosparam param="rate" subst_value="True">$(arg rate)</rosparam>

            <param name="ip" type="string" value="$(arg ip)" />

            <rosparam param="toggle_drone" subst_value="True">$(arg toggle_drone)</rosparam>
            
            <rosparam param="vehicle_name" subst_value="True">$(arg vehicle_name)</rosparam>

            <rosparam param="pose_topic" subst_value="True">$(arg pose_topic)</rosparam>
            <rosparam param="pose_frame" subst_value="True">$(arg pose_frame)</rosparam>

            <rosparam param="vehicle_base_frame" subst_value="True">$(arg vehicle_base_frame)</rosparam>

            <rosparam param="sensor_imu_enable" subst_value="True">$(arg sensor_imu_enable)</rosparam>
            <rosparam param="sensor_imu_name" subst_value="True">$(arg sensor_imu_name)</rosparam>
            <rosparam param="sensor_imu_topic" subst_value="True">$(arg sensor_imu_topic)</rosparam>
            <rosparam param="sensor_imu_frame" subst_value="True">$(arg sensor_imu_frame)</rosparam>
      </node>

      <node pkg="rosbag" type="record" name="rosbag_record" args="record -o /mnt/c/Users/woute/airsim_route_only -a"/>
</launch>
